tide_site.module

```diff
@@ -662,3 +714,55 @@ function tide_site_search_api_index_items_alter(IndexInterface $index, array &$i
 function tide_site_tide_api_jsonapi_custom_query_parameters_alter(&$custom_params) {
   $custom_params[] = 'site';
 }
+
+/**
+ * Implements hook_menu_local_tasks_alter().
+ */
+function tide_site_menu_local_tasks_alter(&$data, $route_name, RefinableCacheableDependencyInterface &$cacheability) {
+  $routes = ['system.admin_content', 'entity.media.collection'];
+  if (in_array($route_name, $routes)) {
+    if (isset($data['tabs'][0]['views_view:view.files.page_1'])) {
+      unset($data['tabs'][0]['views_view:view.files.page_1']);
+    }
+    if (isset($data['tabs'][0]['scheduled_update.admin'])) {
+      unset($data['tabs'][0]['scheduled_update.admin']);
+    }
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function tide_site_form_views_exposed_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  if (!_has_target_site_fields()) {
+    return;
+  }
+
+  $primary_sites_term_options = [];
+  $storage = Drupal::entityTypeManager()->getStorage('taxonomy_term');
+  $primary_sites_terms = $storage->loadTree('sites', 0, 1, TRUE);
+  foreach ($primary_sites_terms as $term) {
+    $primary_sites_term_options[$term->id()] = $term->getName();
+  }
+  $options = ['All' => t('- Any -')] + $primary_sites_term_options;
+  $form['field_node_primary_site_target_id']['#options'] = $options;
+}
+
+/**
+ * Check if field_node_primary_site and field_node_site in the database.
+ */
+function _has_target_site_fields() {
```

Comment:
@vincent-gao It's worth adding `_tide` to the beginning of this function name to keep the naming convention consistent and make it clear where it's come from if another module were to use it.
```suggestion
function _tide_has_target_site_fields() {
```

Reply:
@vincent-gao have you considered `getSupportedEntityTypes()` from the `tide_site.helper` service?